### PR TITLE
Add unit tests for daemon util functions

### PR DESCRIPTION
**Pull request by GautamBytes**
### What this PR does / why we need it

This PR introduces a comprehensive unit test suite for the utility functions in `pkg/daemon/util/util.go`. These functions, which handle critical image reference parsing and environment variable reading, previously lacked test coverage.

Adding these tests improves code quality, verifies the correctness of the parsing logic against multiple edge cases, and helps prevent future regressions.

### Which issue(s) this PR fixes

In Support of Issue #2074 

### Special notes for your reviewer

The test suite covers all exported functions, including:
- `NodeName`
- `ParseRegistry`
- `ParseRepositoryTag`
- `NormalizeImageRefToNameTag`
- `NormalizeImageRef`

### CODE REVIEW

File: pkg/daemon/util/util_test.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+func TestParseRegistry(t *testing.T) {
+	cases := []struct {
+		name      string
+		imageName string
+		want      string
+	}{
+		{name: "image with docker.io", imageName: "docker.io/library/ubuntu", want: "docker.io"},
+		{name: "image with gcr.io", imageName: "gcr.io/project/image", want: "gcr.io"},
+		{name: "image with registry and port", imageName: "myregistry.com:5000/user/image", want: "myregistry.com:5000"},
+		{name: "image without registry", imageName: "ubuntu", want: "ubuntu"},
```
Comment (furykerry) [inline pkg/daemon/util/util_test.go]: maybe for image without registry, ParseRegistry should return docker.io

Comment (GautamBytes) [reply pkg/daemon/util/util_test.go]: did the suggested change!

**Review by furykerry — APPROVED** [review]
/lgtm
/approve
